convert_hf_to_gguf.py

```diff
@@ -1990,16 +1993,59 @@ def set_vocab_tekken(self):
         self.gguf_writer.add_add_bos_token(True)
         self.gguf_writer.add_add_eos_token(False)
 
-        script_dir = Path(__file__).parent
-        template_path = script_dir / "models/templates/unsloth-mistral-Devstral-Small-2507.jinja"
-        with open(template_path, "r", encoding="utf-8") as f:
-            template = f.read()
-            self.gguf_writer.add_chat_template(template)
+    def set_vocab(self):
+        if self.is_mistral_format:
+            return self._set_vocab_mistral()
+
+        path_tekken_json = self.dir_model / "tekken.json"
+        path_tokenizer_json = self.dir_model / "tokenizer.json"
+        if path_tekken_json.is_file() and not path_tokenizer_json.is_file():
+            self._set_vocab_mistral()
+
+            script_dir = Path(__file__).parent
+            template_path = script_dir / "models/templates/unsloth-mistral-Devstral-Small-2507.jinja"
+            with open(template_path, "r", encoding="utf-8") as f:
+                template = f.read()
+                self.gguf_writer.add_chat_template(template)
```

Comment:
I'm wondering if we can also move this to `_set_vocab_mistral()`, as for now GGUF converted via `--mistral-format` won't have any templates built-in, thus it won't be usable out-of-the box with tools based on llama.cpp.

IMO forcing people to use both python (to format the chat) and llama.cpp at the same time may not be a good user experience, so having built-in jinja template should still be a requirement.

Also self-note: one thing missing from this code, we should make sure we are using the correct tekken version for the given model, maybe as a `assert tekken_json["config"]["version"] == "v7"`. But I can add it later.